lib/images.rb

```diff
@@ -0,0 +1,51 @@
+class Image
+
+  def line_is_image_tag(str)
+    str.include?('![')
+  end
+
+  def has_opt_title(str)
+    str.include?('"') || str.include?("'")
+  end
+  
+  def split_reverse(str)
+    str.split('(').map!{|x| x}.reverse.join
+  end
+  
+  def add_alt_tag(str)
+    str.sub!('![',  ' alt="')
+    str.sub!(']', '">')
+  end
+
+  def add_src_tag(str)
+    str.insert(0, '<img src="')
+    str.sub!(')', '"')
+  end
+
+  def add_title_tag(str)
+    str.include?('"') ? str.sub!('"', '" title="') : str.sub!("'", "title='") 
+    str.sub!(')', ' ') 
+  end
+  
+  def convert_image_tag(str)
+    if line_is_image_tag(str) == true && has_opt_title(str) == true
+      str = split_reverse(str)
+      add_alt_tag(str)
+      add_title_tag(str)
+      add_src_tag(str)
+      str
+    elsif line_is_image_tag(str) == true 
+      str = split_reverse(str)
```

Comment:
this can be a method as well

Try to keep all methods under 8 lines, a few 10-14 liners is OK, but always aim for short methods :)